api/python/cellxgene_census/src/cellxgene_census/_release_directory.py

```diff
@@ -11,6 +11,10 @@
 import requests
 from typing_extensions import TypedDict
 
+from ._util import _uri_join
+
+SUPPORTED_PROVIDERS = ["S3", "file"]
+
 """
 The following types describe the expected directory of Census builds, used
 to bootstrap all data location requests.
```

Comment:
Please add a trivial example to the comment that shows both mirrors.json and release.json (just for ease of reading the code).  I'd also add a sentence or two about how the two are used to resolve the actual Census location.

It is also worth noting that the JSON data MAY contain other fields, which should be ignored, ie.., are private or undefined in this interface.

Reply:
Also, please document which fields are going to be deprecated/removed in a future version.  Ideally document both the old and new API usage pattern (or add a unit test to the same effect)

Reply:
`test_open_soma_works_if_no_relative_uri_specified` is the test for the old behavior. I will add some docs as requested.

Reply:
follow-up idea: maybe it is time for a short design doc on release.json/mirrors.json to live in the docs directory?  That would be a good place to document the data model assumptions, etc.